File: steps/contact_form_steps.py
import pytest
from pytest_bdd import given, when, then
from page_objects.contact_page import ContactPage


@pytest.fixture
def contact_page(page):
    return ContactPage(page)


@given('I am on the "Contact" page')
def visit_contact_page(contact_page):
    contact_page.visit("https://automationintesting.online/#/contact")


@when('I fill out the contact form with valid data')
def fill_form_with_valid_data(contact_page):
    contact_page.fill_form(
        name="Test User",
        email="test@example.com",
        subject="Test Subject",
        message="This is a test message."
    )


@when('I fill out the contact form with invalid email')
def fill_form_with_invalid_data(contact_page):
    contact_page.fill_form(
        name="Test User",
        email="invalid-email",
        subject="Test Subject",
        message="This is a test message."
    )


@when('I submit the form')
def submit_form(contact_page):
    contact_page.submit_form()


@then('I should see a success message')
def verify_success_message(contact_page):
    assert "Thanks for getting in touch" in contact_page.get_success_message()


@then('I should see an error message')
def verify_error_message(contact_page):
    assert "Please provide a valid email address" in contact_page.get_error_message()

File: steps/contactFormSteps.ts

import { Given, When, Then } from '@cucumber/cucumber';
import { Page } from 'playwright';
import { ContactPage } from '../pageObjects/ContactPage';

let page: Page;
let contactPage: ContactPage;

Given('I am on the {string} page', async (pageName: string) => {
  await page.goto('https://automationintesting.online/' + pageName.toLowerCase());
  contactPage = new ContactPage(page);
});

When('I fill out the form with valid data', async () => {
  await contactPage.fillForm({
    name: 'Test User',
    email: 'test@example.com',
    subject: 'Testing',
    message: 'This is a test message.'
  });
});

When('I fill out the form with invalid email', async () => {
  await contactPage.fillForm({
    name: 'Test User',
    email: 'invalid-email',
    subject: 'Testing',
    message: 'This is a test message.'
  });
});

When('I submit the form', async () => {
  await contactPage.submitForm();
});

Then('I should see a success message', async () => {
  const successMessage = await contactPage.getSuccessMessage();
  expect(successMessage).toContain('Thanks for getting in touch');
});

Then('I should see an error message', async () => {
  const errorMessage = await contactPage.getErrorMessage();
  expect(errorMessage).toContain('Please provide a valid email address');
});
